satpy/tests/reader_tests/test_cmsaf_claas.py

```diff
@@ -183,15 +184,70 @@ def file_handler(self, fake_file):
         from satpy.readers.cmsaf_claas2 import CLAAS2
         return CLAAS2(fake_file, {}, {})
 
+    @pytest.mark.parametrize(
+        ("ds_name", "expected"),
+        [
+            ("ctt", xr.DataArray([[280, 290], [300, 310]], dims=("y", "x"))),
+            ("cph", xr.DataArray([[0, 1], [2, 0]], dims=("y", "x"))),
+        ]
+    )
+    def test_get_dataset(self, file_handler, ds_name, expected):
+        """Test dataset loading."""
+        dsid = make_dataid(name=ds_name)
+        ds = file_handler.get_dataset(dsid, {})
+        xr.testing.assert_allclose(ds, expected)
+
+    def test_start_time(self, file_handler, start_time):
+        """Test start time property."""
+        assert file_handler.start_time == start_time
+
+    def test_end_time(self, file_handler):
+        """Test end time property."""
+        assert file_handler.end_time == datetime.datetime(2085, 8, 13, 13, 15)
+
+class TestAreaDef:
+    """Test the area of both CLAAS2 and CLAAS3 products"""
+    @pytest.fixture(params=[3636, 3712])
+    def grid_size(self, request):
+        return request.param
+
+    @pytest.fixture
+    def fake_file(self, tmp_path, start_time, grid_size):
+        """Write a fake dataset to file.
+
+        Only add y dimension, so that the file is small but we can still check the
+        real-life area definition.
+        """
+        filename = tmp_path / "CPPin20140101001500305SVMSG01MD.nc"
+        ds = netCDF4.Dataset(filename, mode="w")
+        ds.createDimension("y", grid_size)
+        ds.setncattr("time_coverage_start", start_time.strftime("%Y%m%dT%H%MZ"))
```

Comment:
I used the wrong time format here, this should fix the problem:

```suggestion
        ds.setncattr("time_coverage_start", start_time.strftime(""%Y-%m-%dT%H:%M:%SZ""))
```

(`fromisoformat` doesn't support all iso formats in Python-3.10; and this is the format in the CLAAS files)

Reply:
Yeah! We found out that independently on our debugging session with @jequierz :grin:

Reply:
It was also weird that the test passed with Python 3.12 locally. I guess the NetCDF4 library has added some datetime handling to newer versions to make it work even with the "wrong" time string.